plugins/org.polarsys.capella.scenario.editor.embeddededitor/src/org/polarsys/capella/scenario/editor/embeddededitor/commands/XtextToDiagramCommands.java

```diff
@@ -76,14 +77,19 @@ public static void process(Scenario scenario, EmbeddedEditorView embeddedEditorV
       // Project project;
       BlockArchitecture blockArchitecture = BlockArchitectureExt.getRootBlockArchitecture(scenario);
 
-      doEditingOnParticipants(scenario, blockArchitecture, participants);
+      doEditingOnParticipants(scenario, blockArchitecture, participants, messages);
 
       doEditingOnMessages(scenario, blockArchitecture, messages);
+
+      // refresh
+      EmbeddedEditorView eeView = XtextEditorHelper.getActiveEmbeddedEditorView();
+      Scenario scenarioDiagram = EmbeddedEditorInstance.getAssociatedScenarioDiagram();
+      DiagramToXtextCommands.process(scenarioDiagram, eeView);
```

Comment:
No need to refresh the changes to diagram as the user shall press the save button to do that. These lines shall be removed.